### PR TITLE
Minor fixes for macOS build with gcc

### PR DESCRIPTION
**Pull request by barracuda156**
1. Ensure BE is defined: add a canonical macro supported by gcc and clang.
2. Fix a build error with gcc.

P. S. `-fobjc-exceptions` is supported with clang as well, right?

### CODE REVIEW

**Comment by radioactiveman** [comment]
Thanks for the patches. Could you please test the patches below? `WORDS_BIGENDIAN` should be defined globally, but this is not the case. Neither for audacious nor audacious-plugins.

**audacious**

```diff
diff --git a/src/config.h.meson b/src/config.h.meson
index 4081a16c2..48fb9cb78 100644
--- a/src/config.h.meson
+++ b/src/config.h.meson
@@ -5,6 +5,8 @@
 #mesondefine EXPORT
 #mesondefine PLUGIN_SUFFIX
 #mesondefine VALGRIND_FRIENDLY
+#mesondefine WORDS_BIGENDIAN
+#mesondefine BIGENDIAN
 
 #define PACKAGE_VERSION VERSION
 #define ICONV_CONST
```

**audacious-plugins**

```diff
diff --git a/meson.build b/meson.build
index 8b57a1f62..2434344df 100644
--- a/meson.build
+++ b/meson.build
@@ -118,6 +118,9 @@ conf.set_quoted('VERSION', meson.project_version())
 conf.set('PACKAGE_VERSION', meson.project_version())
 if host_machine.endian() == 'big'
   conf.set10('WORDS_BIGENDIAN', true)
+  conf.set10('BIGENDIAN', true)
+else
+  conf.set10('BIGENDIAN', false)
 endif
 
 
diff --git a/src/config.h.meson b/src/config.h.meson
index d191e7715..d1e978812 100644
--- a/src/config.h.meson
+++ b/src/config.h.meson
@@ -3,6 +3,8 @@
 #mesondefine COPYRIGHT
 #mesondefine EXPORT
 #mesondefine PLUGIN_SUFFIX
+#mesondefine WORDS_BIGENDIAN
+#mesondefine BIGENDIAN
 
 #define PACKAGE_VERSION VERSION
 #define ICONV_CONST
```

File: src/mac-media-keys/meson.build [hunk 1]
```diff
@@ -9,6 +9,7 @@ shared_module('mac-media-keys',
   mac_media_keys_sources,
   dependencies: [audacious_dep],
   name_prefix: '',
+  objc_args: ['-std=c99', '-fobjc-exceptions'],
```
Comment (radioactiveman) [inline src/mac-media-keys/meson.build:12]: Is it really necessary to specify the c99 std? What is the error message with GCC? 

And for consistency I think it's better to update `mac-media-keys/Makefile` for Autotools as well.

**Comment by barracuda156** [comment]
@radioactiveman Errors without flags passed, gcc 14.1.0:

```
FAILED: src/mac-media-keys/mac-media-keys.dylib.p/SPInvocationGrabbing.m.o 
ccache /opt/local/bin/gcc-mp-14 -Isrc/mac-media-keys/mac-media-keys.dylib.p -Isrc/mac-media-keys -I../audacious-plugins-4.4/src/mac-media-keys -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -O0 -g -ffast-math -Wtype-limits -Wno-stringop-truncation -fvisibility=hidden -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_multimedia_audacious-plugins/audacious-plugins/work/build/src/config.h -pipe -Os -arch ppc -MD -MQ src/mac-media-keys/mac-media-keys.dylib.p/SPInvocationGrabbing.m.o -MF src/mac-media-keys/mac-media-keys.dylib.p/SPInvocationGrabbing.m.o.d -o src/mac-media-keys/mac-media-keys.dylib.p/SPInvocationGrabbing.m.o -c ../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m: In function '-[SPInvocationGrabber runInBackground]':
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m:50:14: error: '-fobjc-exceptions' is required to enable Objective-C exception syntax
   50 |         @try {
      |              ^
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m: In function '-[SPInvocationGrabber printBacktrace]':
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m:102:9: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  102 |         for(int x = 3; x < frameCount; x++) {
      |         ^~~
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m:102:9: note: use option '-std=c99', '-std=gnu99', '-std=c11' or '-std=gnu11' to compile your code
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m: In function '-[NSObject(SPInvocationGrabbing) onMainAsync:]':
../audacious-plugins-4.4/src/mac-media-keys/SPInvocationGrabbing.m:134:5: warning: right-hand operand of comma expression has no effect [-Wunused-value]
  134 |     grabber.waitUntilDone = !async;
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[203/363] ccache /opt/local/bin/gcc-mp-14 -Isrc/mac-media-keys/mac-media-keys.dylib.p -Isrc/mac-media-keys -I../audacious-plugins-4.4/src/mac-media-keys -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -O0 -g -ffast-math -Wtype-limits -Wno-stringop-truncation -fvisibility=hidden -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_multimedia_audacious-plugins/audacious-plugins/work/build/src/config.h -pipe -Os -arch ppc -MD -MQ src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o -MF src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o.d -o src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o -c ../audacious-plugins-4.4/src/mac-media-keys/SPMediaKeyTap.m
FAILED: src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o 
ccache /opt/local/bin/gcc-mp-14 -Isrc/mac-media-keys/mac-media-keys.dylib.p -Isrc/mac-media-keys -I../audacious-plugins-4.4/src/mac-media-keys -I/opt/local/include -fdiagnostics-color=always -Wall -Winvalid-pch -O0 -g -ffast-math -Wtype-limits -Wno-stringop-truncation -fvisibility=hidden -include /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_multimedia_audacious-plugins/audacious-plugins/work/build/src/config.h -pipe -Os -arch ppc -MD -MQ src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o -MF src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o.d -o src/mac-media-keys/mac-media-keys.dylib.p/SPMediaKeyTap.m.o -c ../audacious-plugins-4.4/src/mac-media-keys/SPMediaKeyTap.m
../audacious-plugins-4.4/src/mac-media-keys/SPMediaKeyTap.m: In function '-[SPMediaKeyTap shouldInterceptMediaKeyEvents]':
../audacious-plugins-4.4/src/mac-media-keys/SPMediaKeyTap.m:174:22: error: '-fobjc-exceptions' is required to enable Objective-C exception syntax
  174 |         @synchronized(self) {
      |                      ^
ninja: build stopped: subcommand failed.
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_multimedia_audacious-plugins/audacious-plugins/work/build" && /opt/local/bin/ninja -j6 -v 
Exit code: 1
```

**Comment by barracuda156** [comment]
Strictly speaking, both flags are only needed for `mac-media-keys` plug-in. If it is disabled, `plugins` build fine without flags being added.

**Comment by barracuda156** [comment]
Ah, sorry, the patch is already being applied only to that plugin anyway :)

**Comment by radioactiveman** [comment]
Both issues should be fixed now with https://github.com/audacious-media-player/audacious-plugins/commit/5bf91040a8c88611dc530effd5b6fc7f5c82f144 and https://github.com/audacious-media-player/audacious-plugins/commit/c175d8b9cece8a53d86ce54fc252454287814e9e.
The real issue here was that Meson didn't include `WORDS_BIGENDIAN` in `config.h` and therefore `LOCAL_BE` was never defined.

```c
#ifdef WORDS_BIGENDIAN
# define LOCAL_BE
#else
# define LOCAL_LE
#endif
```

Thanks again for your patches and the pull request. :+1:

**Comment by barracuda156** [comment]
@radioactiveman Thank you!

**Comment by barracuda156** [comment]
@radioactiveman Could we return to this issue? Turned out that while 4.4 with my patches works correctly (normal sound output via SDL), 4.4.1 without patches is broken (only hiss). Both tested on the same system, everything else equal, I just switch between port versions. 4.3.1 and 4.4 are good, 4.4.1 and 4.4.2 are broken. From our side 4.4.1 dropped my patches.

**Comment by barracuda156** [comment]
@radioactiveman Looks like something different has broken though. Apparently a coincidence with patches dropping. I rebuilt 4.4.2 with my BE patch added back, and nothing changed, output does not work.

**Comment by barracuda156** [comment]
@radioactiveman Hmm, in fact reverting both https://github.com/audacious-media-player/audacious/commit/9e51ffa0c0ec79b537cc970158d51557273970ea and https://github.com/audacious-media-player/audacious-plugins/commit/c175d8b9cece8a53d86ce54fc252454287814e9e plus restoring my https://github.com/audacious-media-player/audacious-plugins/pull/166/commits/343d20c6ece2d48dba06a8ba4f474499a8a3a1f0 fixed sound output on 4.4.2.

**Comment by radioactiveman** [comment]
Do you use Autotools or Meson? And is `LOCAL_BE` or `LOCAL_LE` defined on your system?
The upper ifdef you have changed shouldn't matter if `WORDS_BIGENDIAN` is set correctly.

https://github.com/audacious-media-player/audacious-plugins/blob/239bd61b10314434233f669899b01da208ee0d6c/src/xsf/desmume/types.h#L99-L109

**Pull request closed without merging**